### ISSUE TITLE
HiDive Does Not Detect or Download English Subtitle File For Japanese Audio (and other issues)

### ISSUE THREAD

**Issue by LunaGNUisance** [yt-dlp/yt-dlp#408]
## Checklist

- [X] I'm reporting a broken site support issue
- [X] I've verified that I'm running yt-dlp version **2021.06.09**
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [X] I've searched the bugtracker for similar bug reports including closed ones
- [X] I've read bugs section in FAQ

## Verbose log

```
$ yt-dlp --cookies cookies.txt --referer "https://www.hidive.com/stream/nana/s01e001" https://www.hidive.com/stream/nana/s01e001 --skip-download --list-subs -v

[debug] Command-line config: ['--cookies', 'cookies.txt', '--referer', 'https://www.hidive.com/stream/nana/s01e001', 'https://www.hidive.com/stream/nana/s01e001', '--skip-download', '--list-subs', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] yt-dlp version 2021.06.08 
[debug] Python version 3.9.5 (CPython 64bit) - Linux-5.12.9-arch1-1-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4, ffprobe 4.4
[debug] Proxy map: {}
[debug] [HiDive] Extracting URL: https://www.hidive.com/stream/nana/s01e001
[HiDive] nana/s01e001: Downloading JSON metadata
[HiDive] nana/s01e001: Downloading m3u8 information
[HiDive] nana/s01e001: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
Available subtitles for nana/s01e001:
Language Formats
en       vtt, vtt

```

## Description

While it is possible to download a subtitle file, this subtitle file is not for the Japanese audio; it appears to be only for the English audio, with subtitles for things such as the lyrics for a Japanese song in this case. The vast majority of the audio does not have subtitles, but on the site, when playing the Japanese audio, there are subtitles for every line. It seems that this subtitle file is not being picked up when using the `--all-subtitles`, `--write-subtitles`, or `--list-subs` options.

This issue only seems to apply to series that have both an English dub and sub, as other series with only Japanese audio were able to download the correct subtitle file.

This issue was originally brought up for youtube-dl over a year ago, but was closed as off-topic without resolution: https://github.com/ytdl-org/youtube-dl/issues/23830

HiDive requires a premium account in order to watch episodes, though occasionally they will let you watch free episodes here: https://www.hidive.com/free-episodes

### Downloads Mismatched Audio/Video by Default
Additionally, this command can be used to preferentially download the Japanese audio/video format over the English audio/video format, as it suffers from the same issue as Animelab's extractor (https://github.com/yt-dlp/yt-dlp/issues/347):
`yt-dlp --add-metadata --all-subs --cookies cookies.txt -f "[format_id^=ja_hv_or-hls]/[format_id^=en_hv_or-hls]"`

### Must Use `--referer` Option
Also, another issue with HiDive is that you need to use the `--referer $URL` option before the episode URL for it to work. as discussed in this issue: https://github.com/ytdl-org/youtube-dl/issues/28575

**Comment by pukkandan** [comment]
Which country is this accessible in?

**Comment by LunaGNUisance** [comment]
> Which country is this accessible in?

I believe it is predominantly the United States, though I'm able to access some of the series from Australia. Couldn't find much information on the website about what countries it supports.

**Comment by pukkandan** [comment]
#958 should have fixed this, closing since OP hasn't replied

**Comment by pukkandan** [comment]
@ocrbonk Please post your verbose log here and let us know whether you are willing to share account details or not (https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed)

**Comment by OCRBonk** [comment]
> @OCRBonk Please post your verbose log here and let us know whether you are willing to share account details or not (https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed)

I would be willing to potentially share account info.
I've tried a few configs, but it never seems to be able to pull subs.

> 
C:\Program Files (x86)\MPC-HC>"C:\Program Files (x86)\MPC-HC\hidive download.bat
"

C:\Program Files (x86)\MPC-HC>yt-dlp.exe --check-all-formats -f - --embed-subs -
-sub-lang en --sub-format vtt -u buttwipinglord@yahoo.com -p qP3cBWOerLwwxap --a
dd-metadata  -v https://www.hidive.com/stream/legend-of-the-galactic-heroes/s01e
001
[debug] Command-line config: ['--check-all-formats', '-f', '-', '--embed-subs',
'--sub-lang', 'en', '--sub-format', 'vtt', '-u', 'PRIVATE', '-p', 'PRIVATE', '--
add-metadata', '-v', 'https://www.hidive.com/stream/legend-of-the-galactic-heroe
s/s01e001']
[debug] Portable config file: C:\Program Files (x86)\MPC-HC\yt-dlp.conf
[debug] Portable config: []
[debug] Encodings: locale cp1252, fs utf-8, out utf-8 (No ANSI), err utf-8 (No A
NSI), pref cp1252
[debug] yt-dlp version 2021.12.27 [6223f67] (win_exe)
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg 2021-12-27-git-617452ce2c-full_build-www.gyan.dev (
setts), ffprobe 2021-12-27-git-617452ce2c-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[HiDive] None: Downloading webpage
[HiDive] Logging in
[debug] [HiDive] Extracting URL: https://www.hidive.com/stream/legend-of-the-gal
actic-heroes/s01e001
[HiDive] legend-of-the-galactic-heroes/s01e001: Downloading JSON metadata
[HiDive] legend-of-the-galactic-heroes/s01e001: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7),
vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext,
aext, hasaud, source, id
[HiDive] legend-of-the-galactic-heroes/s01e001: Downloading webpage
[info] Testing format ja_hv_or-6583
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 146
[download] Destination: C:\Program Files (x86)\MPC-HC\tmpibhhayep.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format ja_hv_or-3000
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 146
[download] Destination: C:\Program Files (x86)\MPC-HC\tmpm8uyqbtf.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format ja_hv_or-1585
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 146
[download] Destination: C:\Program Files (x86)\MPC-HC\tmp6iy8ahg1.tmp
[download] 100% of 10.00KiB in 00:00
[info] Testing format ja_hv_or-1089
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 146
[download] Destination: C:\Program Files (x86)\MPC-HC\tmpp9j_3eeb.tmp
[download] 100% of 10.00KiB in 00:00
[info] Available formats for legend-of-the-galactic-heroes/s01e001:
ID            EXT RESOLUTION |   TBR PROTO  | VCODEC        VBR ACODEC    ABR MO
RE INFO
--------------------------------------------------------------------------------
---------
ja_hv_or-1089 mp4 640x360    | 1089k m3u8_n | avc1.4d001e 1089k mp4a.40.2  0k [j
a] hv, or
ja_hv_or-1585 mp4 854x480    | 1585k m3u8_n | avc1.4d001f 1585k mp4a.40.2  0k [j
a] hv, or
ja_hv_or-3000 mp4 1280x720   | 3000k m3u8_n | avc1.640029 3000k mp4a.40.2  0k [j
a] hv, or
ja_hv_or-6583 mp4 1920x1080  | 6583k m3u8_n | avc1.640029 6583k mp4a.40.2  0k [j
a] hv, or

Enter format selector: ja_hv_or-6583
[info] legend-of-the-galactic-heroes/s01e001: Downloading 1 format(s): ja_hv_or-
6583
[debug] Invoking downloader on "https://www.hidive.com/manifest/child/df9ef1e2cf
84821da9ae70dd158183d8a76fd640/LGH_s01e001_hv_or_na_ja_xx_HLS1080p_16x9_00_v.m3u
8?#="
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 146
[download] Destination: legend-of-the-galactic-heroes_s01e001 [legend-of-the-gal
actic-heroes_s01e001].mp4
[download] 100% of 587.02MiB in 02:00
[debug] ffprobe command line: ffprobe -hide_banner -show_format -show_streams -p
rint_format json "file:legend-of-the-galactic-heroes_s01e001 [legend-of-the-gala
ctic-heroes_s01e001].mp4"
[FixupM3u8] Fixing MPEG-TS in MP4 container of "legend-of-the-galactic-heroes_s0
1e001 [legend-of-the-galactic-heroes_s01e001].mp4"
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:legend-o
f-the-galactic-heroes_s01e001 [legend-of-the-galactic-heroes_s01e001].mp4" -c co
py -map 0 -dn -f mp4 "-bsf:a" aac_adtstoasc -movflags "+faststart" "file:legend-
of-the-galactic-heroes_s01e001 [legend-of-the-galactic-heroes_s01e001].temp.mp4"

[EmbedSubtitle] There aren't any subtitles to embed
[Metadata] Adding metadata to "legend-of-the-galactic-heroes_s01e001 [legend-of-
the-galactic-heroes_s01e001].mp4"
[debug] ffmpeg command line: ffmpeg -y -loglevel "repeat+info" -i "file:legend-o
f-the-galactic-heroes_s01e001 [legend-of-the-galactic-heroes_s01e001].mp4" -map
0 -dn -c copy -metadata "title=legend-of-the-galactic-heroes/s01e001" -metadata
"purl=https://www.hidive.com/stream/legend-of-the-galactic-heroes/s01e001" -meta
data "comment=https://www.hidive.com/stream/legend-of-the-galactic-heroes/s01e00
1" -metadata "show=legend-of-the-galactic-heroes" -metadata "season_number=1" -m
etadata "episode_id=Episode 1" -metadata "episode_sort=1" -movflags "+faststart"
 "file:legend-of-the-galactic-heroes_s01e001 [legend-of-the-galactic-heroes_s01e
001].temp.mp4"

C:\Program Files (x86)\MPC-HC>

**Comment by wamasi** [comment]
They're specifying `--sub-lang "en"` and it's not picking up what's available for that video, "english-subs".

Using @OCRBonk's URL from their log, when I supply `--sub-lang "en.*"`, `--sub-lang "all"` or `--sub-lang "english-subs"` I am able to grab them. I only get `[EmbedSubtitle] There aren't any subtitles to embed` when I don't use regex or spell it out to match what's available.

For HiDive reference, I've seen English subs under the following names:
english-subs
english-caps

**Comment by pukkandan** [comment]
@OCRBonk, please confirm. Or, show the output of `yt-dlp --list-subs -Uv URL`

**Comment by OCRBonk** [comment]
Interesting. I've tried --list-subs before and it always says "___ Has no Subtitles"
https://pastebin.com/f2R7XsVX
Same again.

When I try the `--sub-lang "all"` still get same problem
https://pastebin.com/BKUnLB2N

**Comment by aforst** [comment]
This seems to be an issue once again. I cannot do the following:

- List subtitles at all
- List video formats
- Specify English subtitles (I don't appear to be able to get this with "--sub-lang "en.*", --sub-lang "all" or --sub-lang "english-subs" as written by @wamasi above)

This means that I can't get Japanese audio with English subs consistently right now. This is the command I'm using that currently gets Japanese audio and random subs (currently Spanish):

`yt-dlp -u $username -p $password -f "(b*[height<=1080][language*=ja])" --embed-subs --all-subs --embed-metadata --no-mtime --verbose --download-archive /mnt/h/archive-hidive.txt -P $downloadpath/ --external-downloader aria2c --batch-file $hidiveurls >> $logoutput`

If I specify English subs, it just says there aren't any available:
`[EmbedSubtitle] There aren't any subtitles to embed`

Output for listing available subtitles:
`yt-dlp -u [username here] -p [password here] --list-subs -Uv https://www.hidive.com/stream/urusei-yatsura/s01e002
[debug] Command-line config: ['-u', 'PRIVATE', '-p', 'PRIVATE', '--list-subs', '-Uv', 'https://www.hidive.com/stream/urusei-yatsura/s01e002']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (pip)
[debug] Python 3.10.6 (CPython x86_64 64bit) - Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2023.05.07, secretstorage-3.3.1, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[HiDive] None: Downloading webpage
[HiDive] Logging in
ERROR: [HiDive] urusei-yatsura/s01e002: Unable to extract profile_id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/aj/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 691, in extract
    self.initialize()
  File "/home/aj/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 597, in initialize
    self._perform_login(username, password)
  File "/home/aj/.local/lib/python3.10/site-packages/yt_dlp/extractor/hidive.py", line 50, in _perform_login
    profile_id = self._search_regex(r'<button [^>]+?data-profile-id="(\w+)"', login_webpage, 'profile_id')
  File "/home/aj/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 1242, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)`

Output for showing available files:
`yt-dlp -u [username here] -p [password here] -F -Uv h
ttps://www.hidive.com/stream/urusei-yatsura/s01e002
[debug] Command-line config: ['-u', 'PRIVATE', '-p', 'PRIVATE', '-F', '-Uv', 'https://www.hidive.com/stream/urusei-yatsura/s01e002']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d] (pip)
[debug] Python 3.10.6 (CPython x86_64 64bit) - Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: Cryptodome-3.11.0, brotli-1.0.9, certifi-2023.05.07, secretstorage-3.3.1, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[HiDive] None: Downloading webpage
[HiDive] Logging in
ERROR: [HiDive] urusei-yatsura/s01e002: Unable to extract profile_id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/aj/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 691, in extract
    self.initialize()
  File "/home/aj/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 597, in initialize
    self._perform_login(username, password)
  File "/home/aj/.local/lib/python3.10/site-packages/yt_dlp/extractor/hidive.py", line 50, in _perform_login
    profile_id = self._search_regex(r'<button [^>]+?data-profile-id="(\w+)"', login_webpage, 'profile_id')
  File "/home/aj/.local/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 1242, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)`

**Comment by OCRBonk** [comment]
Here's the command that seems to be working for me now. It does pull and embed EN subs successfully.
`yt-dlp.exe -f - --embed-subs --sub-lang "all" --sub-format vtt -u **-p *** --add-metadata  -v Hi-dive url here`
